dependabot/generate-build-time-stats.py

```diff
@@ -0,0 +1,181 @@
+    wait_cycles = 0
+    while pending:
+        if wait_cycles < MAX_WAIT_CYCLES:
+            time.sleep(SLEEP_INTERVAL)
+            pending, passing, failing_pr_checks = check_pending_pr_checks(dist_repo, pr)
+            if not pending:
+                if len(failing_pr_checks) > 0:
+                    passing = all(check.startswith('codecov') for check in failing_pr_checks)
+                if passing:
+                    if(pr.mergeable_state != 'dirty'):
+                        try:
+                            pr.merge()
+                            ref.delete()
+                            log_message = "[Info] Automated build-time-stats generation PR is merged. PR: " + pr.html_url
+                            print(log_message)
+                        except Exception as e:
+                            print("[Error] Error occurred while merging automated build time stats generation PR " , e)
+                    else:
+                        print("[Info] Automated build-time-stats generation PR is unmerged due to conflicts." + "\n" +\
+                                "Please visit <" + pr.html_url + "|the build page> for more information")
+                else:
+                    print("[Info] Automated build-time-stats generation PR PR has failed checks." + "\n" +\
+                        "Please visit <" + pr.html_url + "|the build page> for more information")
+                    pr.edit(state = 'closed')
+                    ref.delete()
+            else:
+                wait_cycles += 1
+        else:
+            print("[Info] Automated build-time-stats generation PR is unmerged due to pr checks timeout." + "\n" +\
+                "Please visit <" + pr.html_url + "|the build page> for more information")
+            break
+
+
+def create_pull_request(repo):
+    try:
+        pull_request_title = "[Automated] Add build time statistics files"
+        created_pr = repo.create_pull(
+                        title=pull_request_title,
+                        body='[Automated] Add build time statistics files',
+                        head="temp-build-time-data",
+                        base="build-time-data"
+                    )
+        return created_pr
+    except Exception as e:
+        print("[Error] Error occurred while creating pull request for branch build-time-data.", e)
+        sys.exit(1)
+
+
+def check_pending_pr_checks(repo, pr):
+    passing = True
+    pending = False
+
+    failed_pr_checks = []
+    pull_request = repo.get_pull(pr.number)
+    sha = pull_request.head.sha
+    for pr_check in repo.get_commit(sha=sha).get_check_runs():
+        if pr_check.status != 'completed':
+            pending = True
+            break
+        elif pr_check.conclusion == 'success':
+            continue
+        else:
+            failed_pr_checks.append(pr_check.name)
+            passing = False
+    return pending, passing, failed_pr_checks
+
+def approve_pr(module, auto_merge_pull_requests, pr_number):
```

Comment:
we can import this from `utils` and use here instead of repeating here. Ex: https://github.com/ballerina-platform/ballerina-release/blob/master/dependabot/update_lang_version.py#L54

Reply:
Done